Sources/MongoSwift/BSON/Document+Codable.swift

```diff
@@ -46,9 +46,10 @@ extension Document: Codable {
         for key in container.allKeys {
             let k = key.stringValue
             if try container.decodeNil(forKey: key) {
-                output[k] = nil
+                try output.setValue(forKey: k, to: nil)
             } else {
-                output[k] = try container.decode(AnyBSONValue.self, forKey: key).value
+                let val = try container.decode(AnyBSONValue.self, forKey: key).value
```

Comment:
This line was broken up into two because the idea of having a nested `try` inside of an outer `try` seemed less readable and less explicit about what points in our code can actually throw.